Bottom: 6cdd653d7aef88e3264ff019e81e47e3b3cabae3
Top:    61b9e5a902ac47a912f6e9342cff71adc5d784c9
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2008-10-13 23:37:32 +0100

Add profiling to file load







---

diff --git a/src/file.c b/src/file.c
index 738188a..35bfb57 100644
--- a/src/file.c
+++ b/src/file.c
@@ -350,6 +350,10 @@ LoadPCB (char *Filename)
 {
   PCBTypePtr newPCB = CreateNewPCB (False);
   Boolean units_mm;
+  clock_t start, end;
+  double elapsed;
+
+  start = clock();
 
   /* new data isn't added to the undo list */
   if (!ParsePCB (newPCB, Filename))
@@ -380,6 +384,10 @@ LoadPCB (char *Filename)
 	  CreateDefaultFont ();
 	}
 
+  end = clock();
+  elapsed = ((double) (end - start)) / CLOCKS_PER_SEC;
+  printf ("Loading file %s took %f\n", Filename, elapsed);
+
       /* clear 'changed flag' */
       SetChangedFlag (False);
       PCB->Filename = MyStrdup (Filename, "LoadPCB()");
